From 23099818523dd9ef3a2ec988e4fd294ff128a18e Mon Sep 17 00:00:00 2001
From: "Kevin F. Haggerty" <haggertk@lineageos.org>
Date: Mon, 19 Nov 2018 06:20:16 -0700
Subject: [PATCH 6/9] msm8974-common: sepolicy: Resolve system_app denials

* avc: denied { call } for comm=4173796E635461736B20233130
  scontext=u:r:system_app:s0 tcontext=u:r:storaged:s0 tclass=binder
  permissive=0
* avc: denied { read } for name="pagetypeinfo" dev="proc"
  ino=4026543036 scontext=u:r:system_app:s0
  tcontext=u:object_r:proc_pagetypeinfo:s0 tclass=file permissive=0
* avc: denied { open } for comm=4173796E635461736B202332
  name="pagetypeinfo" dev="proc" ino=4026543036
  scontext=u:r:system_app:s0 tcontext=u:object_r:proc_pagetypeinfo:s0
  tclass=file permissive=0

Change-Id: I7f90debc146b3e87ba4a3625a7a9f74e80a21dfc
---
 sepolicy/common/system_app.te  | 1 +
 sepolicy/private/system_app.te | 1 +
 sepolicy/sepolicy.mk           | 3 +++
 3 files changed, 5 insertions(+)
 create mode 100644 sepolicy/private/system_app.te

diff --git a/sepolicy/common/system_app.te b/sepolicy/common/system_app.te
index 9aa08bc..b0cc99d 100644
--- a/sepolicy/common/system_app.te
+++ b/sepolicy/common/system_app.te
@@ -1,2 +1,3 @@
+allow system_app proc_pagetypeinfo:file r_file_perms;
 allow system_app sysfs_mdnie:dir search;
 allow system_app sysfs_mdnie:file rw_file_perms;
diff --git a/sepolicy/private/system_app.te b/sepolicy/private/system_app.te
new file mode 100644
index 0000000..39dedf5
--- /dev/null
+++ b/sepolicy/private/system_app.te
@@ -0,0 +1 @@
+binder_call(system_app, storaged)
diff --git a/sepolicy/sepolicy.mk b/sepolicy/sepolicy.mk
index 18f7153..adfcfab 100644
--- a/sepolicy/sepolicy.mk
+++ b/sepolicy/sepolicy.mk
@@ -17,5 +17,8 @@
 include device/qcom/sepolicy-legacy/sepolicy.mk
 
 # Board specific SELinux policy variable definitions
+BOARD_PLAT_PRIVATE_SEPOLICY_DIR += \
+    device/samsung/msm8974-common/sepolicy/private
+
 BOARD_SEPOLICY_DIRS += \
     device/samsung/msm8974-common/sepolicy/common
-- 
2.17.1

